### PR TITLE
ERMS-2508

### PR DESCRIPTION
**Pull request by cjaegle**

### CODE REVIEW

File: app/grails-app/services/com/k_int/kbplus/FactService.groovy [hunk 1]
```diff
@@ -433,6 +436,86 @@ class FactService {
...
+          if (! months.contains(m))
+            months.add(m)
+        }
+      }
+      Date min = ranges*.availFrom.min()
+      if (subscription?.startDate){
+        if (min < subscription.startDate){
+          min = subscription.startDate
+          months.removeAll { it ->
+            it < java.time.YearMonth.parse(subscription.startDate.toString().substring(0,7))
```
Comment (croqueGrec09) [inline app/grails-app/services/com/k_int/kbplus/FactService.groovy:477]: unnecessary qualified reference

File: app/grails-app/services/com/k_int/kbplus/FactService.groovy [hunk 2]
```diff
@@ -433,6 +436,86 @@ class FactService {
...
+      ])
+      List months = []
+      ranges.each(){
+        getMonthsOfDateRange(it.availFrom, it.availTo).each { m ->
+          if (! months.contains(m))
+            months.add(m)
+        }
+      }
+      Date min = ranges*.availFrom.min()
+      if (subscription?.startDate){
```
Comment (croqueGrec09) [inline app/grails-app/services/com/k_int/kbplus/FactService.groovy:473]: Can subscription be null here?

File: app/grails-app/services/com/k_int/kbplus/FactService.groovy (hunk 1 above)
Comment (cjaegle) [reply app/grails-app/services/com/k_int/kbplus/FactService.groovy:477]: Replaced it with an import

File: app/grails-app/services/com/k_int/kbplus/FactService.groovy (hunk 2 above)
Comment (cjaegle) [reply app/grails-app/services/com/k_int/kbplus/FactService.groovy:473]: Yes, it can be null. But I should rework this method, when possible.

**Pull request merged**